iree/compiler/Conversion/LinalgToLLVM/LinalgBufferizePass.cpp

```diff
@@ -0,0 +1,447 @@
+/// template instantiating one pattern for each linalg::LinalgOp.
+LogicalResult convertAnyLinalgOp(OpBuilder &b, linalg::LinalgOp op,
+                                 BlockAndValueMapping &bvm) {
+  OpBuilder::InsertionGuard g(b);
+  b.setInsertionPoint(op);
+  Location loc = op.getLoc();
+  SmallVector<Value, 2> newInputBuffers;
+  newInputBuffers.reserve(op.getNumInputs());
+  for (Value v : op.getInputs()) newInputBuffers.push_back(bvm.lookup(v));
+  SmallVector<Value, 2> newOutputBuffers(op.getOutputBuffers().begin(),
+                                         op.getOutputBuffers().end());
+
+  if (failed(allocateBuffersForResults(b, loc, op, newOutputBuffers, bvm)))
+    assert(false);
+
+  // Delegate to the linalg generic pattern.
+  if (auto genericOp = dyn_cast<linalg::GenericOp>(op.getOperation())) {
+    finalizeBufferAllocation(b, genericOp, newInputBuffers, newOutputBuffers,
+                             bvm);
+    return success();
+  }
+
+  finalizeBufferAllocation(b, op, newInputBuffers, newOutputBuffers, bvm);
+  return success();
+}
+
+// Extract int64_t values from the assumed ArrayAttr of IntegerAttr.
+static SmallVector<int64_t, 4> extractFromI64ArrayAttr(Attribute attr) {
+  return llvm::to_vector<4>(llvm::map_range(
+      attr.cast<ArrayAttr>(),
+      [](Attribute a) -> int64_t { return a.cast<IntegerAttr>().getInt(); }));
+}
+
+static MemRefType getMemrefTypeForTensor(
+    RankedTensorType tensorType, ArrayRef<AffineMap> affineMapComposition = {},
+    unsigned memorySpace = 0) {
+  return MemRefType::get(tensorType.getShape(), tensorType.getElementType(),
+                         affineMapComposition, memorySpace);
+}
+
+// TODO: this will go away once integrated at the flow level.
+SymbolRefAttr getBinding(Operation *op) {
+  return llvm::TypeSwitch<Operation *, SymbolRefAttr>(op)
+      .Case<IREE::HAL::InterfaceLoadTensorOp, IREE::HAL::InterfaceStoreTensorOp,
+            IREE::HAL::InterfaceLoadTensorTileOp,
+            IREE::HAL::InterfaceStoreTensorTileOp>(
+          [](auto op) { return op.binding(); })
+      .Default([](auto op) {
+        llvm_unreachable("Expected op with binding");
+        return SymbolRefAttr();
+      });
+}
+
+// Create the placeholder op for the backing buffer.
+// TODO: evolve to target flow ops directly with additional shape annotation.
+IREE::PlaceholderOp createPlaceholderOp(OpBuilder &b, Location loc,
+                                        Operation *op, Value rankedTensor,
+                                        IREE::HAL::InterfaceBindingOp bindingOp,
+                                        bool typeErase = false) {
+  assert(bindingOp);
+  OpBuilder::InsertionGuard g(b);
+  b.setInsertionPointToStart(
+      &op->getParentOfType<FuncOp>().getBlocks().front());
+
+  // Get the corresponding memref type from the tensor type.
+  auto tensorType = rankedTensor.getType().cast<RankedTensorType>();
+  auto bufferType = getMemrefTypeForTensor(tensorType);
+
+  if (typeErase)
```

Comment:
Nit: Please add braces for conditional that spans many lines. 
Also IREE style (which is Google style) is to always add braces.

Reply:
ok, converted them all